pkg/controllers/gpu/controller.go

```diff
@@ -242,32 +242,54 @@ func removeVolumeMount(volumeMounts []v1.VolumeMount, name string) []v1.VolumeMo
 	return newVolumeMounts
 }
 
-func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (updated bool) {
-	dp := rawObj.(*devicepluginv1.GpuDevicePlugin)
-
-	if ds.Spec.Template.Spec.Containers[0].Image != dp.Spec.Image {
-		ds.Spec.Template.Spec.Containers[0].Image = dp.Spec.Image
-		updated = true
-	}
+func processInitcontainer(ds *apps.DaemonSet, dp *devicepluginv1.GpuDevicePlugin) bool {
```

Comment:
nitpick: Other parts use 'initContainer'. Do we need to unify them?

Reply:
Unified.